packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -984,7 +984,8 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this._connected = connected;
 
         if (changeOfState) {
-           this.replayPendingStates();
+            this.pendingStateManager.replayInitialStates();
```

Comment:
This looks suspicious. In a sense that we likely can't afford to allow connected events to propagate through the system until we fully observed initial local ops, which can take time as we need to process not just locally saved ops, but also remote ops. I'd expect that once we are done with process of consuming local state, container would not not need nay more to know that it started that way, and most of the code not to have any knowledge of initial state.

This is based on our earlier discussions, I have not finished looking through it to see if design matches what I think it is.